conans/test/functional/printer_test.py

```diff
@@ -0,0 +1,161 @@
+import unittest
+import os
+from conans.test.utils.tools import TestClient, TestServer
+
+conanfile_libA = """
+from conans import ConanFile
+
+class AConan(ConanFile):
+    name = "aaa"
+    version = "0.1"
+"""
+
+conanfile_libB = """
+from conans import ConanFile
+
+class BConan(ConanFile):
+    name = "bbb"
+    version = "0.1"
+    requires = "aaa/0.1@lasote/channel"
+"""
+
+conanfile_libC = """
+from conans import ConanFile
+
+class CConan(ConanFile):
+    name = "ccc"
+    version = "0.1"
+    requires = "bbb/0.1@lasote/channel"
+    exports = "txt_file.txt"
+
+"""
+
+txt_file = """
+    example file
+    """
+
+test_profile = """
```

Comment:
Packages don't have settings. Profile irrelevant, don't use it

Reply:
The profile is used to test `Printer.print_profile()` output

Reply:
This is part of the confusion that this tests is testing other unrelated things to this fix.
Refactoring and improving tests is something to do, but I'd say better in a different PR, referencing the other tests that should cover it and are not doing it.